src/main/java/inventorysetups/InventorySetupsPersistentDataManager.java

```diff
@@ -108,14 +106,30 @@ public void updateConfig(boolean updateSetups, boolean updateSections)
 	{
 		if (updateSetups)
 		{
-			List<InventorySetupSerializable> issList = new ArrayList<>();
+			// Rather than escaping the name to pick a config key, instead use hash of the name. Benefits:
+			//   Changes to the escaping function or supported characters in the name string don't affect config keys.
+			//   Standardizes length; preventing overflow issues from extremely long names
+			//   Keeps json as the single source of truth for names; other plugins won't try to grab names from config keys.
+			final String wholePrefix = ConfigManager.getWholeKey(CONFIG_GROUP, null, CONFIG_KEY_SETUPS_V3_PREFIX);
+			final List<String> oldSetupKeys = configManager.getConfigurationKeys(wholePrefix);
+			Set<String> oldSetupHashes = oldSetupKeys.stream().map(k -> k.substring(wholePrefix.length())).collect(Collectors.toSet());
+
+			// Fast non-cryptographic hash with relatively short output. Could use murmur3_32_fixed instead; but being extra cautious of collisions.
+			final HashFunction hf = Hashing.murmur3_128();

```

Comment:
Should we instead make this a member variable of the data manager class? If this is not a singleton then we are creating a brand new hashing function each time. Seems like we can just create one and keep it forever.

Reply:
also, a name better than `hf`. `hashFunction` is fine

Reply:
It's a singleton; we could also move it into the class if you want.

Reply:
Yeah why not, pretty minor but helps clear up that it's only being created once even if that is the case already

Reply:
done; set it as `public static`; which is definitely debatable though. lmk if you want me to change that.